public/app/features/browse-dashboards/components/RestoreModal.tsx

```diff
@@ -27,8 +27,14 @@ export const RestoreModal = ({
   const numberOfDashboards = selectedDashboards.length;
 
   useEffect(() => {
-    if (Object.entries(dashboardOrigin).length === 1 && dashboardOrigin[selectedDashboards[0]] !== 'general') {
-      setRestoreTarget(dashboardOrigin[selectedDashboards[0]]);
+    // restoreTarget is used by the folder picker to preselect a folder and therefore unblock the confirm button
+    // if there is only one dashboard selected or all selected dashboards come from the same folder
+    if (
+      dashboardOrigin.length === 1 ||
+      (dashboardOrigin.length > 1 &&
+        dashboardOrigin.every((originalLocation) => originalLocation === dashboardOrigin[0]))
+    ) {
```

Comment:
i think this could even be:
```tsx
    if (
      dashboardOrigin.length > 0 &&
        dashboardOrigin.every((originalLocation) => originalLocation === dashboardOrigin[0])
    ) {
```

Reply:
True, I wasn't sure about this since I was worried that the intention behind it isn't clear enough anymore but I assume the comment explains it fully